torch/_dynamo/guards.py

```diff
@@ -814,15 +936,16 @@ def convert(size_or_stride):
             + list(SYMPY_INTERP.items())
         )
         closure_vars.update(CLOSURE_VARS)
-        py_code = f"""\
-def ___make_guard_fn({','.join(closure_vars.keys())}):
-    return lambda L: {code}
-"""
+
+        unique_code_parts = list(unique(code_parts))
+        make_guard_fn_args = ", ".join(closure_vars.keys())
+        pycode = build_guard_function(unique_code_parts, make_guard_fn_args)
+
         if os.environ.get("TORCHDYNAMO_PRINT_GUARDS", None) == "1":
-            print("GUARDS", code)
+            print("GUARDS", " and ".join(unique_code_parts))
```

Comment:
TBH we should just print the if sled now...

Reply:
Let's do (2); (1) is probably ok too but the outer goop is probably not enlightening. Not (3) because you want to print what actually gets executed.